asciinema/asciicast/__init__.py

```diff
@@ -77,6 +84,14 @@ def open_url(url):
 
         return utf8_reader(body, errors='strict')
 
+    _, encoding = mimetypes.guess_type(url)
+    if encoding == "gzip":
+        with codecs.open(url, mode="r", encoding="gzip") as fh:
+            return StringIO(fh.read().decode("utf-8"))
```

Comment:
This requires loading whole recording into memory, which is incompatible with [streaming features introduced in 2.0](https://blog.asciinema.org/post/two-point-o/). I'd rather keep it consistent, and made gzip/bz2 support streaming.

Something like this (for gzip case) should work I think:

```python
# gzip
return gzip.open(url, mode='rt', encoding='utf-8')

# bz2
return bz2.open(url, mode='rt', encoding='utf-8')
```

https://docs.python.org/3/library/gzip.html
https://docs.python.org/3/library/bz2.html